### DISCUSSION TITLE
Use plt to see the image with model and customize camera and rays

### DISCUSSION THREAD

**Discussion by yangocean-sudo** [mitsuba-renderer/mitsuba3#561] [Q&A] [answered]
Hello, I'm reading the tutorial [https://mitsuba.readthedocs.io/en/stable/src/rendering/scripting_renderer.html], and want to know what is the result before rendering. 
Could you tell me that before sampling, how can I check the original image (The sphere from the XML file and the grid of rays)?
Thank you very much.

**Comment by njroussel** [comment, marked as answer]
I'm not quite sure I understand. Do you want a tool to visualize the scene and rays? We do not provide something like that out of the box.

**Reply by yangocean-sudo** [reply]
Sorry that my question is not straightforward. After setting up a camera, and a set of rays with original points and directions(cam_dir), how can I see the resulting image before the next step(ambient occlusion), like using 
```
image = mi.TensorXf(result, shape=image_res)
import matplotlib.pyplot as plt
plt.imshow(image, cmap='gray'); plt.axis('off');
``` 

(I'm not going to visualize rays)
Or is it not enough to generate an image ? Sorry  I'm new to this field

**Reply by njroussel** [reply]
You can't really visualize anything at that point. For every ray, you want to intersect your scene and then define some metric that you want to capture and write that value to a pixel.
With just a sphere and a ray, what would you want to write to a pixel? Typically you would continue bouncing around your scene up until your hit a light. You might want to read some more about pathtracing (in general) before jumping right into Mitsuba - the learning curve will be less steep.

**Reply by yangocean-sudo** [reply]
Thank you for your reply, I learnt basic knowledge about path tracing. I understand now that the rays are probe rays, not light rays. The light is the emitter in the original XML file, is it correct?

**Reply by njroussel** [reply]
Yes :+1:
